hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -256,8 +256,8 @@ public Result inputSplits(
       HoodieTimeline archivedCompleteTimeline = archivedTimeline.getCommitsTimeline().filterCompletedInstants();
       if (!archivedCompleteTimeline.empty()) {
         Stream<HoodieInstant> instantStream = archivedCompleteTimeline.getInstants();
-        return maySkipCompaction(instantStream)
-            .map(instant -> WriteProfiles.getCommitMetadata(tableName, path, instant, archivedTimeline)).collect(Collectors.toList());
+        return instantStream.map(instant -> WriteProfiles.getCommitMetadata(tableName, path, instant, archivedTimeline))
+          .collect(Collectors.toList());
```

Comment:
You mean for incremental read, the user does not need to read the change log (or only snapshot is needed) ?